File: Backend/Movieverse.Contracts/DataTransferObjects/User/UserDto.cs
using Movieverse.Domain.ValueObjects;

namespace Movieverse.Contracts.DataTransferObjects.User;

public sealed class UserDto
{
	public string UserName { get; set; } = null!;
	public string Email { get; set; } = null!;
	public Information Information { get; set; } = null!;
	public Guid? AvatarPath { get; set; }
	public bool EmailConfirmed { get; set; }
	public Guid? PersonId { get; set; }
	public bool Banned { get; set; }
	public bool CanChangePassword { get; set; }
}

File: Frontend/movieverse/src/core/dtos/user/userDto.ts
export interface InformationDto {
  firstName: string | undefined;
  lastName: string | undefined;
  age: string;
}

export interface UserDto {
  userName: string;
  email: string;
	phoneNumber: string;
	information: InformationDto;
	avatarPath: string | undefined;
	emailConfirmed: boolean;
	personId: string | undefined;
}